### ISSUE TITLE
Impossibility to have only direct (or local) -used- traits via an FM3Trait

### ISSUE THREAD

**Issue by enwiro** [moosetechnology/Fame#23]
It is impossible to have only the traits used directly (or locally) via an FM3Trait. That is to say that the "traits" method of an FM3Trait returns its entire trait usage chain.

For example, for the FM3Trait: FAST-Core-Model.TNullPointerLiteral, its method returns 4 traits going back to Moose-Query.TEntityMEtaLevelDependency.
![image](https://user-images.githubusercontent.com/58632394/119463778-5ebf0400-bd42-11eb-8734-21f4710dceaf.png)

While the equivalent Pharo trait, via "implementingClass traits", returns only the direct FASTTLiteral trait. But be careful, it returns a Pharo trait and not an FM3Trait.

![image](https://user-images.githubusercontent.com/58632394/119464713-4f8c8600-bd43-11eb-9fb2-887d0c9eaf2b.png)

In conclusion, the problem is not to be able to access the FM3Traits used by another FM3Trait without going through the Pharo trait and using a hack.

**Comment by badetitou** [comment]
# First try

## Idea
We try to implement this feature by modifying method ```FMMetaModelBuilder>>#processClass:ifPragmaAbsent:```, by using method `traits` instead of `allGeneratedTraits`.

## Problem

But we got a problem when resetting all metamodels.
The problem comes a hidden dependency between the MooseCore package and Moose-Query package

## Possible solutions

One idea would be to clarify the dependency between the two packages : MooseCore and Moose-Query

**Comment by NicolasAnquetil** [comment]
Moose-Core depending on Moose-Query ?
This does not seem right :-(

**Comment by badetitou** [comment]
The dependency is:

The `MooseAbstractGroup` class uses the `TMooseGroupDependencyQueries` trait.
I suppose it is correct.
If so, we should clearly define that Moose-Core depends on Moose-Query
